apollo-federation/src/link/context_spec_definition.rs

```diff
@@ -30,13 +39,26 @@ impl ContextSpecDefinition {
         }
     }
 
-    pub(crate) fn context_directive_name_in_schema(
+    pub(crate) fn context_directive_definition<'schema>(
+        &self,
+        schema: &'schema FederationSchema,
+    ) -> Result<&'schema Node<DirectiveDefinition>, FederationError> {
+        self.directive_definition(schema, &CONTEXT_DIRECTIVE_NAME_IN_SPEC)?
+            .ok_or_else(|| {
+                SingleFederationError::Internal {
+                    message: "Unexpectedly could not find context spec in schema".to_owned(),
+                }
+                .into()
+            })
+    }
+
+    pub(crate) fn context_directive_arguments<'doc>(
         &self,
-        schema: &FederationSchema,
-    ) -> Result<Name, FederationError> {
-        Ok(self
-            .directive_name_in_schema(schema, &CONTEXT_DIRECTIVE_NAME_IN_SPEC)?
-            .unwrap_or(CONTEXT_DIRECTIVE_NAME_DEFAULT))
```

Comment:
Bugs:
- This only returns `None` when there's no `@core`/`@link` in the schema, which should never be the case for supergraph schemas. (Generally we've been using functions like `context_directive_definition()` instead, which gets you the name/definition and also asserts the definition exists.)